tools/snippet-testing/language_handler/base_handler.rb

```diff
@@ -53,20 +56,43 @@ def execute_with_suppressed_output(command, file)
       exit_code = check_process_success(pid, command)
       wout.close
       werr.close
-      success = exit_code.zero? && language_conditional(rout)
+      output = rout.read
+      success = exit_code.zero? && check_stdout(output) && language_conditional(rout)
 
       if success
         puts "success [#{lang_cname}]".green
       else
         puts "failure [#{lang_cname}]".red
         error_message = rerr.read
+
+        if @test_output
+          error_message += "\ndiff:".red
+          error_message += "\n  sample: #{ @current_sample }"
+          error_message += "\n  output: #{ output.chomp }"
+        end
+
         ErrorLogger.instance.add_error(file, error_message, ErrorLogger::ERROR)
       end
+
       rout.close
       rerr.close
       success
     end
 
+    def check_stdout(output)
+      return true unless @test_output
+
+      output_xml =  "#{File.dirname(@input_file)}/output/#{File.basename(@input_file).split('.').first}.xml"
+      assert_output = false
+      File.open(output_xml, 'r') do |file|
+        @current_sample = file.read.gsub(/\r|\n|\t/, '')
```

Comment:
Yeah, we were not actually comparing with regex but sanitizing a little to make the comparison given the XMLs seemed to have a straightforward syntax while avoiding having dependencies on another XML parser library like Nokogiri. This actually worked for the time being but I consider you are right, not all of the frameworks have the same rules when it comes to hash order handling which might affect the output order and/or generate other issues. Thanks for the tips up!